### PR TITLE
Dynamic filtering

### PR DESCRIPTION
**Pull request by TheCovenant**

### CODE REVIEW

File: app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapFilterStateTypes.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*
+Each attribute of a Feature with type 'number' should have these 
+properties on filterStateOfType
+*/
+export interface filterStateTypes{
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapFilterStateTypes.ts]: CamelCase name, same as file name.

File: app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapState.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
+import { filterStateTypes} from "./MapFilterStateTypes";
 
 export interface MapState {
   keplerGlInstanceRegistered: boolean;
   features: MapFeature[];
   typesVisibility: {[index: string]: boolean};
-  featureTypesFilters: {[featureType: string]: MapFilterState};
+  featureTypesFilters: {[featureType: string]: {[attribute: string]: filterStateTypes}};
+  filterCounter: number;
+  attributeIds: number;
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapState.ts]: If this is a counter please call it a counter.

File: app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapState.ts [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import {MapFeature} from "./MapFeature";
-import {MapFilterState} from "./MapFilterState";
+import { filterStateTypes} from "./MapFilterStateTypes";
 
 export interface MapState {
   keplerGlInstanceRegistered: boolean;
   features: MapFeature[];
   typesVisibility: {[index: string]: boolean};
-  featureTypesFilters: {[featureType: string]: MapFilterState};
+  featureTypesFilters: {[featureType: string]: {[attribute: string]: filterStateTypes}};
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapState.ts]: You can use numbers as object indices if that's what this is [attribute: number]. I'm not sure.

Comment (TheCovenant) [reply app/geo/gui/src/ts/twxplore/gui/geo/states/map/MapState.ts]: I want this to be a map of maps.
E.g. featureTypesFilters['Transmission']['frequency'] which returns a {min, max, idx} filterStateType object

File: app/geo/gui/src/ts/twxplore/gui/geo/components/selectionPanel/SelectionPanel.tsx [hunk 4]
```diff
@@ -39,6 +41,8 @@ const SelectionPanelImpl: React.FunctionComponent = () => {
     dispatch(changeTypeVisibility(typeEnum));
   };
 
+  const logger : ConsoleLogger = React.useContext(LoggerContext)
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/selectionPanel/SelectionPanel.tsx]: It's a Logger, not a ConsoleLogger.

File: lib/ts/base/src/ts/twxplore/lib/base/util/logging/LoggerContext.ts [hunk 5]
```diff
@@ -1,5 +1,9 @@
 import {ConsoleLogger} from "./ConsoleLogger";
 import {Logger} from "./Logger";
 import * as React from "react";
+import {Environment} from "../../Environment";
+import {NopLogger} from "./NopLogger";
 
-export const LoggerContext = React.createContext<Logger>(new ConsoleLogger());
+export const LoggerContext = React.createContext<Logger>(
+  Environment.development ? new ConsoleLogger() : new NopLogger()
```
Comment (gordom6) [inline lib/ts/base/src/ts/twxplore/lib/base/util/logging/LoggerContext.ts]: You need a similar statement in main.tsx.
Also, presumably you'll want to pass the logger in main.tsx to your store instances so the reducer has a logger.

Comment (gordom6) [reply lib/ts/base/src/ts/twxplore/lib/base/util/logging/LoggerContext.ts]: The line above is just the default value if no logger is specified, but you see in main.tsx that it always is, so this line could probably have stayed new ConsoleLogger() (or even new NopLogger).

**Pull request merged**